src/graph/ford_fulkerson.rs

```diff
@@ -1,141 +1,295 @@
-/*
-The Ford-Fulkerson algorithm is a widely used algorithm to solve the maximum flow problem in a flow network. The maximum flow problem involves determining the maximum amount of flow that can be sent from a source vertex to a sink vertex in a directed weighted graph, subject to capacity constraints on the edges.
+//! The Ford-Fulkerson algorithm is a widely used algorithm to solve the maximum flow problem in a flow network.
+//!
+//! The maximum flow problem involves determining the maximum amount of flow that can be sent from a source vertex to a sink vertex
+//! in a directed weighted graph, subject to capacity constraints on the edges.
 
-The following is simple idea of Ford-Fulkerson algorithm:
-
-   1. Start with initial flow as 0.
-   2. While there exists an augmenting path from the source to the sink:
-       i. Find an augmenting path using any path-finding algorithm, such as breadth-first search or depth-first search.
-
-       ii. Determine the amount of flow that can be sent along the augmenting path, which is the minimum residual capacity along the edges of the path.
-
-       iii. Increase the flow along the augmenting path by the determined amount.
-    3.Return the maximum flow.
-
-*/
 use std::collections::VecDeque;
 
-const V: usize = 6; // Number of vertices in graph
+/// Enum representing the possible errors that can occur when running the Ford-Fulkerson algorithm.
+#[derive(Debug, PartialEq)]
+pub enum FordFulkersonError {
```

Comment:
What if the `graph` is not a square matrix?

Reply:
Just need to add an error variant to handle this case.